### ISSUE TITLE
improvement: proxy_requests for sensu::check

### ISSUE THREAD

**Issue by exeral** [sensu/sensu-puppet#637]
Hello,

**sensu::check** can be improved with a new attribute: **proxy_requests**

Since [sensu version 0.28.0](https://sensuapp.org/docs/0.28/overview/changelog.html#core-v0-28-0), there is a new feature called: [proxy check requests](https://sensuapp.org/docs/0.29/reference/checks.html#proxy-requests-attributes)

Today, there is a workaround by using custom attributes

```

sensu::check { "remote_http-uchiwa":
    command             => "PATH=/opt/sensu/embedded/bin/:\$PATH check-http.rb -u http://:::address::::3000",
    occurrences         => 2,
    interval            => 300,
    refresh             => 600,
    low_flap_threshold  => 20,
    high_flap_threshold => 60,
    standalone          => false,
    subscribers         => 'poller',
    custom              => {
        'proxy_requests' => {
            'client_attributes' => {
                'subscriptions' => 'eval: value.include?("infra-monitoringwebui")'
            }
        }
    },
}
```

could be improved by a builtin attribute for **sensu::check**

example:
```
sensu::check { "remote_http-uchiwa":
    command             => "PATH=/opt/sensu/embedded/bin/:\$PATH check-http.rb -u http://:::address::::3000",
    occurrences         => 2,
    interval            => 300,
    refresh             => 600,
    low_flap_threshold  => 20,
    high_flap_threshold => 60,
    standalone          => false,
    subscribers         => 'poller',
    proxy_requests	=> {
    	'client_attributes' => {
        	'subscriptions' => 'eval: value.include?("infra-monitoringwebui")'
        }
    },
}
```

since 'client_attributes' seems a hardcoded and mandatory keyword, it could be included as default
resulting as a simpler attributes like:
```
    proxy_requests	=> {
    	'subscriptions' => 'eval: value.include?("infra-monitoringwebui")'
    },
```

sadly, I have no idea how to implement it in sensu/lib/puppet !

Kind regards

**Comment by ghoneycutt** [comment]
@crayfishx this should be very similar to https://github.com/sensu/sensu-puppet/commit/f83e3bfc28f5e23c883422e508c478e6343ea4bf where I added a few parameters to match the API.

**Comment by ghoneycutt** [comment]
Thanks for the great explanation @exeral 

@jeffmccune has implemented this and it is released in v2.21.0
